components/ws-manager/pkg/manager/monitor.go

```diff
@@ -916,6 +922,7 @@ func (m *Monitor) finalizeWorkspaceContent(ctx context.Context, wso *workspaceOb
 		backupError error
 		gitStatus   *csapi.GitStatus
 	)
+	t := time.Now()
```

Comment:
Would it be better that calculate the start time at finalizeWorkspaceContent and have a defer function to calculate the end time? For example:
```go
func (m *Monitor) finalizeWorkspaceContent(ctx context.Context, wso *workspaceObjects) {
    t := time.Now()
    ...
    tpe, err := wso.WorkspaceType()
    if err != nil {
        tracing.LogError(span, err)
        log.WithError(err).Warn("cannot determine workspace type - assuming this is a regular")
        tpe = api.WorkspaceType_REGULAR
    }
    defer func() {
        wsType := api.WorkspaceType_name[int32(tpe)]
	hist, errHist := m.manager.metrics.finalizeTimeHistVec.GetMetricWithLabelValues(wsType)
	if errHist != nil {
		log.WithError(errHist).WithField("type", wsType).Warn("cannot get finalize time histogram metric")
	}
	hist.Observe(time.Since(t).Seconds())
    }()
    ...
}
```

Reply:
Yeah, I thought of that, but that would catch incorrect metrics in case of this `return` statement:
https://github.com/gitpod-io/gitpod/blob/054bd0be5eb1f9e3fff153cf49093c0f424cda35/components/ws-manager/pkg/manager/monitor.go#L929-L933
I only want to catch timing for workspaces where we actually did some work.